CRM/Mailing/BAO/Query.php

```diff
@@ -274,13 +274,16 @@ public static function whereClauseSingle(&$values, &$query) {
 
       case 'mailing_name':
         $value = addslashes($value);
-        if ($wildcard) {
-          $value = "%$value%";
-          $op = 'LIKE';
-        }
 
-        $query->_where[$grouping][] = "civicrm_mailing.name $op '$value'";
-        $query->_qill[$grouping][] = "Mailing Namename $op \"$value\"";
+        $name = 'name';
+        $query->_where[$grouping][] = CRM_Contact_BAO_Query::buildClause("civicrm_mailing.$name", $op, $value, 'String');
+        list($op, $value) = CRM_Contact_BAO_Query::buildQillForFieldValue('CRM_Mailing_DAO_Mailing', $name, $value, $op);
```

Comment:
We haven't made it generic to the other forms yet - but we started making the qill handling generic - would need some moving around 

https://github.com/civicrm/civicrm-core/commit/bf9932666686b8b3fc2240ad11fa8509ed41550e

Reply:
Also note that we have a getFields() function on Contribution_BAO_Query    $fields = self::getFields(); & are working towards using that metadata for the various bits of value